### PR TITLE
New post: "Accelerating PyTorch Transformers with Intel Sapphire Rapids, part 1"

### PR DESCRIPTION
**Pull request by juliensimon**
Distributed training with Intel Sapphire Rapids, IPEX and oneCCL.

Part 2 will cover inference.

### CODE REVIEW

**Review by VioletteLepercq — APPROVED** [review]

**Review by jeffboudier — APPROVED** [review]
Super cool post! Once the public pricing is available it would be super cool to do a cost comparison with GPU.

File: transformers-on-intel-sapphire-rapids-cpus.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+
+## Sapphire Rapids: New Instructions for Deep Learning
+
+The Sapphire Rapids architecture introduces new instructions to accelerate DL workloads:
+
+* [AVX512-FP16](https://en.wikipedia.org/wiki/AVX-512#FP16) brings new instructions dedicated to 16-bit floating point (FP16) data. For example, the VFMADD instruction can multiply two FP16 values and add the result to a third FP16 value, aka 'multiply and add', a common operation used by DL models at inference time.
+
+* The Intel Advanced Matrix Extensions ([AMX](https://en.wikipedia.org/wiki/Advanced_Matrix_Extensions)) are brand new instructions to accelerate matrix multiplication, an operation central to training DL models on data batches. AMX
+ supports both Brain Floating Point ([BF16](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format)) and 8-bit integer (INT8) values, enabling ML practitioners to accelerate both mixed-precision and quantization-aware training jobs.
```
Comment (echarlaix) [inline transformers-on-intel-sapphire-rapids-cpus.md:53]: I would remove the mention of quantization aware training, as it simulates the effects of quantization on both weights and activations, and usually results in more computations / longer training time

File: transformers-on-intel-sapphire-rapids-cpus.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+```
+python run_qa.py --model_name_or_path distilbert-base-uncased \
+--dataset_name squad --do_train --do_eval --per_device_train_batch_size 32 \
+--num_train_epochs 1  --output_dir /tmp/debug_squad/ \
+--use_ipex --bf16 --no_cuda
+```
+
+No need to let the job run to completion, We just run for a minute to make sure that all dependencies have been correctly installed. This also gives us a baseline for single-instance training: 1 epoch takes about **2 hours and 15 minutes**. For reference, we clocked the same job on a comparable Ice Lake instance (`c6i.8xlarge`) at **6 hours** per epoch. We can already see how beneficial the new instructions are!
+
+Now, let's run distribute the training job on two instances. Given that an `r7iz.8xlarge` instance has 32 vCPUs, We decide to allocate 24 (`OMP_NUM_THREADS`) to run 4 Python processes per node (`NUM_PROCESSES_PER_NODE`). Hence, the total number of Python jobs running on the 2-node cluster is 8 (`NUM_PROCESSES`).
```
Comment (echarlaix) [inline transformers-on-intel-sapphire-rapids-cpus.md:203]: I would replace to something like 
```suggestion
Now, let's run the distributed training job on two instances. Given that an `r7iz.8xlarge` instance has 32 vCPUs, we decide to allocate 24 (`OMP_NUM_THREADS`) to run 4 Python processes per node (`NUM_PROCESSES_PER_NODE`). Hence, the total number of Python jobs running on the 2-node cluster is 8 (`NUM_PROCESSES`).
```

File: transformers-on-intel-sapphire-rapids-cpus.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+Perfect linear scaling on 8 nodes would be 16 minutes and 52 seconds (2 hours and 15 minutes divided by 8). We're extremely close to this ideal value, which shows how scalable this approach is.
+
+## Conclusion
+
+As you can see, training Hugging Face transformers on a cluster of Intel Xeon CPUs is a flexible, scalable and cost-effective solution, especially if you're working with small or medium-sized models and datasets.
+
+Here are some additional resources to help you get started:
+
+* [Intel IPEX](https://github.com/intel/intel-extension-for-pytorch) on Github
+* Hugging Face documentation: "[Efficient training on CPU](https://huggingface.co/docs/transformers/perf_train_cpu)"
```
Comment (echarlaix) [inline transformers-on-intel-sapphire-rapids-cpus.md:253]:
```suggestion
* Hugging Face documentation: [Efficient training on CPU](https://huggingface.co/docs/transformers/perf_train_cpu)
```

**Comment by juliensimon** [comment]
On hold while we clarify the AMX config with Intel

**Review by echarlaix — APPROVED** [review]
It looks great @juliensimon 🔥

**Review by julien-c — COMMENTED** [review]
personal nit: let's use a shorter filename (which will be the post URL)

like `intel-sapphire-rapids.md`

**Comment by juliensimon** [comment]
Sending to Intel for review.

**Pull request merged**